python/pyarrow/tests/test_array.py

```diff
@@ -891,6 +891,14 @@ def test_list_from_arrays(list_array_type, list_type_factory):
         result.validate(full=True)
 
 
+def test_map_labelled():
+    t = pa.map_(pa.field("name", "string", nullable=False), "int64")
+    arr = pa.array([[('a', 1), ('b', 2)], [('c', 3)]], type=t)
+    assert arr.type.key_field == pa.field("name", pa.utf8(), nullable=False)
```

Comment:
Would `assert arr.type == t` capture possible incorrect names as well?

Reply:
I tried and it does not. I think that is intentional, given this test in C++:

https://github.com/apache/arrow/blob/85b276dd13512d6676a0bc2dc703a0f9b6934236/cpp/src/arrow/type_test.cc#L1232-L1238

Reply:
For for this PR (maybe for a new JIRA), but it seems we are not very consistent in this. For example for list type it seems to matter:

```
In [6]: l1 = pa.list_(pa.field("val", pa.int64()))

In [7]: l2 = pa.list_(pa.int64())

In [8]: l1
Out[8]: ListType(list<val: int64>)

In [9]: l2
Out[9]: ListType(list<item: int64>)

In [10]: l1 == l2
Out[10]: False
```

Although for a map type, the name of the key field is maybe less important than the values for a list array

Reply:
According to the spec:

https://github.com/apache/arrow/blob/85b276dd13512d6676a0bc2dc703a0f9b6934236/format/Schema.fbs#L112-L115

I don't see anything about the names for a list. IMO seems like a piece of metadata most people want to ignore. I'll create a JIRA to discuss that.

Reply:
https://issues.apache.org/jira/browse/ARROW-14999